### ISSUE TITLE
[0.37.0+] Uncaught TypeError: Cannot read property 'removeAttribute' of null at applySpanProperties

### ISSUE THREAD

**Issue by vinzloh** [handsontable/handsontable#4988]
### Description

What happens:
on calling instance.updateSettings() with merged rows (total rows >= 30, exceeding viewport), this error occurs:
```
Uncaught TypeError: Cannot read property 'removeAttribute' of null at applySpanProperties
```

What should happen:
No error.
### Steps to reproduce

1. Create mergeCells with total rows >= 30 (exceeds viewport)
2. call handsontable instance.updateSettings()
3.  this triggers the error (more in screenshots on the stack trace)

### Demo

This is a private repo, but I've pinpointed the issue with screenshots of the chrome debugger:

1) My sheet has a merged cells in column 0 totalling 30 rows:

<img width="1143" alt="screen shot 2018-04-02 at 3 17 11 pm" src="https://user-images.githubusercontent.com/5501845/38187547-afa6e1a4-368a-11e8-9910-e91549e83751.png">

because of this, `getCell` returns `-2` which function comments indicate is a row after viewport

2) because of this, TD is `-2` and `getCellAtCoords` returns null:

<img width="696" alt="screen shot 2018-04-02 at 3 18 51 pm" src="https://user-images.githubusercontent.com/5501845/38187638-1c8042ca-368b-11e8-995a-5c3744124c49.png">

3) in caller function, array `mergedCellParentsToClear` receives this `null` in the values it pushed in line 51540:

<img width="710" alt="screen shot 2018-04-02 at 3 19 40 pm" src="https://user-images.githubusercontent.com/5501845/38187696-5d83f1ea-368b-11e8-9e57-3a666d3dd732.png">

4) `mergedCellParentsToClear` is used lower down by `_utils.applySpanProperties` on line `51557-51558`. if working correctly, the `null` value should be a `<td>` DOM element, where `applySpanProperties` function tries to `removeAttribute`:

<img width="707" alt="screen shot 2018-04-02 at 3 20 32 pm" src="https://user-images.githubusercontent.com/5501845/38187805-c299ead0-368b-11e8-931f-aadbafe9994a.png">

but in this case, it throws the error. 

This happens as such: `onUpdateSettings > updatePlugin (mergeCell) > clearCollections > ...`

Why am I updating the settings? Because I want to show a different context menu depending on the row and cell clicked, and the only way I know that can trigger the context menu to update is via `instance.updateSettings({contextMenu:{}})`

any temporary workround by anyone?

### Your environment
* Handsontable version: ^0.38.1
* Browser Name and version: Chrome 65.0.3325.181
* Operating System: Mac Sierra 10.12.6

**Comment by AMBudnik** [comment]
Hi @vinzloh 

It looks like I am missing something. Please take a look at this demo http://jsfiddle.net/handsoncode/oog86g0w/

What should I add/change to get the same issue as you have?

**Comment by vinzloh** [comment]
Hi @AMBudnik,

updated demo: [https://jsfiddle.net/oog86g0w/7/](url)

scroll down a little and click inside the merged cell in column 0. 
you should see the error in the console

**Comment by AMBudnik** [comment]
Thank you for sharing. I have replicated the same behavior.

It looks like it happens cause of the `afterOnCellMouseDown` call in the `20` line. If we delete the `hot` parameter the table works fine.

**Comment by AMBudnik** [comment]
You're right. I have changed the viewport a little bit more and it doe the same as yours.

I thought that we have some issue with changing mergeCells areas via `updateSettings` method, but it looks like it only happens for manually merged cells (src: https://github.com/handsontable/handsontable/issues/3315). However, no matter what I do in the `updateSettings` after the `afterOnCellMouseDown` hook is called the error occur and merging fails.

As you previously wrote the viewport has to be narrow so adding `renderAllRows: true` solves the issue. However, it is not a good practice to use it for larger tables.

**Comment by vinzloh** [comment]
Thanks! I'm now using individual icons in the cell itself, since I can't call `updateSettings` without triggering the error. I might use `v0.36.0` as you suggested, so I don't have to destroy and rerender the whole table on each change, which is very expensive. Really appreciate the temporary workaround!

**Comment by DongShi** [comment]
Just come across this issue, for our case, it happens in the initialization stage where mergeCells is passed as a parameter, and it contains certain TD that falls out of the viewport at initial rendering.

Hope this can be fixed soon, we are evaluating HOT against several other vendors extensively, though HOT by far is the most promising one, this issue is definitely a roadmap blocker to adopting HOT in our production.

Is there a recent release to pick up the fix for this one?

**Comment by AMBudnik** [comment]
Hi @DongShi 

if everything works well the new version with this fix will be published on 16th of May.

**Comment by AMBudnik** [comment]
Hi @vinzloh @DongShi we've fixed the issue in version `3.0.0`
Please let me know if it works for you as well.
